opds_import.py

```diff
@@ -173,6 +176,117 @@ def from_config(cls, _db, collection=None):
             collection=collection
         )
 
+    def _run_self_tests(self, _db, lookup_class=None):
+        """Run self-tests on every eligible Collection.
+
+        :param _db: A database connection.
+        :param lookup_class: Pass in a mock class to instantiate that
+           class as needed instead of MetadataWranglerOPDSLookup.
+        :return: A dictionary mapping Collection objects to lists of
+           SelfTestResult objects.
+        """
+        lookup_class = lookup_class or MetadataWranglerOPDSLookup
+        results = dict()
+
+        # Find all eligible Collections on the system, instantiate a
+        # _new_ MetadataWranglerOPDSLookup for each, and call
+        # its _run_collection_self_tests method.
+        for c in _db.query(Collection):
+            try:
+                metadata_identifier = c.metadata_identifier
+            except ValueError, e:
+                continue
+
+            lookup = lookup_class.from_config(_db, c)
+            for i in lookup._run_collection_self_tests():
+                yield i
```

Comment:
This is a little awkward, because we have to instantiate a bunch of `MetadataWranglerOPDSLookup` objects and call `_run_collection_self_tests` on each one. The advantage is that this preserves the behavior of the top-level method, `run_self_tests` --  instantiate a single object, call `_run_self_tests` on it, and expect to get a stream of `HasSelfTest` objects back.